api/v1alpha1/openapi.yaml

```
@@ -3432,11 +3432,88 @@ components:
         - metadata
         - items
       description: FleetList is a list of Fleets.
+
+    DisruptionAllowance:
+      type: object
+      properties:
+        groupBy:
+          type: array
+          items:
+            type: string
+          description: List of labels to perform grouping for the disruption allowance.
+        minAvailable:
+          type: integer
+          description: The maximum number of unavailable devices allowed during rollout.
+        maxUnavailable:
+          type: integer
+          description: The minimum number of required available devices during rollout.
+      description: DisruptionAllowance defines the level of allowed disruption when rollout is in progress.
+
+    Percentage:
+      type: string
+      format: '^(?:(?:[1-9][0-9]?)|100|0)%$'
+      description: Percentage is the string format representing percentage string.
+
+    Batch:
+      type: object
+      properties:
+        selector:
+          $ref: '#/components/schemas/LabelSelector'
+        successThreshold:
+          $ref: '#/components/schemas/Percentage'
+        limit:
+          oneOf:
+            - $ref: '#/components/schemas/Percentage'
+            - type: integer
+              minimum: 1
+      description: Batch is an element in batch sequence.
+    Duration:
+      type: string
+      pattern: '^[1-9]\d*[smhd]$'
+      description: |
+        The maximum duration allowed for the action to complete. 
+        The duration should be specified as a positive integer 
+        followed by a time unit. Supported time units are:
+        - 's' for seconds
+        - 'm' for minutes
+        - 'h' for hours
+        - 'd' for days
+
+    BatchSequence:
+      type: array
+      minItems: 1
+      items:
+        $ref: '#/components/schemas/Batch'
+      description: BatchSequence defines the list of batches to be executed in sequence.
+
+    RolloutPolicy:
+      type: object
+      required:
+        - deviceSelectionStrategy
+      properties:
+        deviceSelectionStrategy:
+          type: string
+          enum: ['batch']
+          default: 'batch'
+        disruptionAllowance:
+          $ref: '#/components/schemas/DisruptionAllowance'
+        batchSequence:
+          $ref: '#/components/schemas/BatchSequence'
+        defaultSuccessThreshold:
```

a default is something that can't change, what do you think about dropping that prefix?
```suggestion
        successThreshold:
```

---

Each batch can have its own success threshold and this is the default in case it is not specified for a batch.